### PR TITLE
New easyconfigs for Imod 4.7.12 and Imod 4.7.15

### PR DESCRIPTION
**Pull request by valtandor**
New easyconfig for imod, installed via a tcsh script. Requires https://github.com/hpcugent/easybuild-easyblocks/pull/847.

### CODE REVIEW

File: easybuild/easyconfigs/i/IMOD/IMOD-4.7.12-RHEL6-64.eb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+#MODULES
+#-------
+#include_modpath_extensions:Include $MODULEPATH extensions specified by module naming scheme. (default: True)
+#modaliases:		Aliases to be defined in module file (default: {})
+#modextrapaths:		Extra paths to be prepended in module file (default: {})
+#modextravars:		Extra environment variables to be added to module file (default: {})
+#modloadmsg:		Message that should be printed when generated module is loaded (default: {})
+#modtclfooter:		Footer to include in generated module file (Tcl syntax) (default: )
+#moduleforceunload:	Force unload of all modules when loading the extension (default: False)
+#moduleloadnoconflict:	Don't check for conflicts, unload other versions instead  (default: False)
```
Comment (JensTimmerman) [inline easybuild/easyconfigs/i/IMOD/IMOD-4.7.12-RHEL6-64.eb]: this is all commented out, so should this be here? or is this WIP?

**Comment by JensTimmerman** [comment]
jenkins: ok to test

**Comment by boegel** [comment]
Jenkins says no...

**Comment by boegel** [comment]
```
AssertionError: Filename '...IMOD-4.7.12-RHEL6-64.eb' of parsed easyconfig matches expected filename 'IMOD-4.7.12_RHEL6-64.eb'
```

**Comment by valtandor** [comment]
I just tried renaming my easyconfig from IMOD-4.7.12-RHEL6-64.eb to IMOD-4.7.12_RHEL6-64.eb. I still get this assertion:

```
AssertionError: Filename '/data/jenkins/workspace/easybuild-easyconfigs-pr-builder/easybuild/easyconfigs/i/IMOD/IMOD-4.7.12-RHEL6-64.eb' of parsed easyconfig matches expected filename 'IMOD-4.7.12_RHEL6-64.eb'
```

I can look into the code and try to figure out where it's coming from, perhaps, but any tips would be most helpful.

**Comment by valtandor** [comment]
On the other hand, it seems that the problem may have been fixed. Let me know if it still persists or if some other error shows up.

**Comment by boegel** [comment]
@wpoely86: can you help resolve the style issues here (maybe just send @valtandor a PR to fix them, that'll be quickest), and look into submitting a test report so we can get this merged?

**Comment by wpoely86** [comment]
This version of IMOD has disappeared on there are now only cuda version?

**Comment by valtandor** [comment]
This is apparently the case. I'll try to prepare an updated installation.

File: easybuild/easyconfigs/i/IMOD/IMOD-4.7.15_RHEL6-64_CUDA6.0.eb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+# Assuming the system java /usr/java can be used. If not, we can
+# cross that bridge.
+moduleclass = 'vis'
+modextravars = {'IMOD_DIR' : "$root",
+                'IMOD_JAVADIR' : "/usr/java",
+                'IMOD_PLUGIN_DIR' : "$root/lib/imodplug",
+                'FOR_DISABLE_STACK_TRACE' : 1,
+                'IMOD_QTLIBDIR' : "$root/qtlib"}
```
Comment (boegel) [inline easybuild/easyconfigs/i/IMOD/IMOD-4.7.15_RHEL6-64_CUDA6.0.eb]: needs style reformat

File: easybuild/easyconfigs/i/IMOD/IMOD-4.7.15_RHEL6-64_CUDA6.0.eb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+			]
+
+sanity_check_paths = {
+    'files': ['IMOD-linux.sh', 'IMOD-linux.csh', 'installIMOD'],
+    'dirs': ['bin', 'lib']
+}
+
+# Assuming the system java /usr/java can be used. If not, we can
+# cross that bridge.
+moduleclass = 'vis'
```
Comment (boegel) [inline easybuild/easyconfigs/i/IMOD/IMOD-4.7.15_RHEL6-64_CUDA6.0.eb]: this should be the very last line

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/817c7382825241b47703 for a full test report.

File: easybuild/easyconfigs/i/IMOD/IMOD-4.7.15_RHEL6-64_CUDA6.0.eb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+sanity_check_paths = {
+    'files': ['IMOD-linux.sh', 'IMOD-linux.csh', 'installIMOD'],
+    'dirs': ['bin', 'lib']
+}
+
+# Assuming the system java /usr/java can be used. If not, we can
+# cross that bridge.
+modextravars = {
+    'IMOD_DIR'                : "$root",
+    'IMOD_JAVADIR'            : "/usr/java",
```
Comment (boegel) [inline easybuild/easyconfigs/i/IMOD/IMOD-4.7.15_RHEL6-64_CUDA6.0.eb]: why not add Java as a dependency?

File: easybuild/easyconfigs/i/IMOD/IMOD-4.7.15_RHEL6-64_CUDA6.0.eb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    'IMOD_PLUGIN_DIR'         : "$root/lib/imodplug",
+    'FOR_DISABLE_STACK_TRACE' : 1,
+    'IMOD_QTLIBDIR'           : "$root/qtlib"
+}
+modaliases = {'subm' : "submfg $* &"}
+modloadmsg = """Please set the environment variable IMOD_CALIB_DIR if appropriate."""
```
Comment (boegel) [inline easybuild/easyconfigs/i/IMOD/IMOD-4.7.15_RHEL6-64_CUDA6.0.eb]: insert empty line below please

File: easybuild/easyconfigs/i/IMOD/IMOD-4.7.15_RHEL6-64_CUDA6.0.eb [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+# -skip: do not attempt to deploy resource files in /etc
+# -yes: do not prompt for confirmation
+installopts = "-dir %(installdir)s -skip -yes"
+
+# The assumption by the install script is that installdir will be something
+# like /usr/local. So it prefixes the install location with a lot of specific
+# directories that are surplus to requirements.
+postinstallcmds = [	"mv %(installdir)s/%(name)s/* %(installdir)s/.",
+			"rm -r %(installdir)s/%(name)s %(installdir)s/%(namelower)s_%(version)s"
+			]
```
Comment (boegel) [inline easybuild/easyconfigs/i/IMOD/IMOD-4.7.15_RHEL6-64_CUDA6.0.eb]: How sure are you that this doesn't break anything?
The installation prefix doesn't get hardcoded anywhere in the installed files?

also, please fix indent:

```python
postinstallcmds = [
    "mv %(installdir)s/%(name)s/* %(installdir)s/.",
    "rm -r %(installdir)s/%(name)s %(installdir)s/%(namelower)s_%(version)s",
]
```

File: easybuild/easyconfigs/i/IMOD/IMOD-4.7.15_RHEL6-64_CUDA6.0.eb [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+programs used for tomographic reconstruction and for 3D reconstruction of EM
+serial sections and optical sections. The package contains tools for assembling
+and aligning data within multiple types and sizes of image stacks, viewing 3-D
+data from any orientation, and modeling and display of the image files. IMOD
+was developed primarily by David Mastronarde, Rick Gaudette, Sue Held, Jim
+Kremer, Quanren Xiong, and John Heumann at the University of Colorado."""
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+toolchainopts = {'openmp': True}
+dependencies = [('CUDA', '6.0.37')]
```
Comment (boegel) [inline easybuild/easyconfigs/i/IMOD/IMOD-4.7.15_RHEL6-64_CUDA6.0.eb]: please move this down, below `sources` (with an empty line in between)

**Comment by valtandor** [comment]
@boegel: Do you know how I can include the contents of EBROOTJAVA in the modextravars? I tried os.getenv, and that didn't work, and I can't see an example of how to do this in the EasyConfigs repo.

**Comment by JensTimmerman** [comment]
I believe you can just use $EBROOTJAVA (or maybe $JAVA_HOME) here, it will be substituted by the modulecmd (or at least that's what I see on our systems for other modules, e.g. hod
https://github.com/hpcugent/easybuild-easyconfigs/blob/de378fd89e4116e1ca8af955ffb7d81dbf58be32/easybuild/easyconfigs/h/hanythingondemand/hanythingondemand-3.0.0-cli.eb

**Comment by JensTimmerman** [comment]
Using this "\$JAVA_HOME" generate a modulefile containing e.g.
`setenv		 HOD_BATCH_WORKDIR $VSC_SCRATCH/hod `
which means this path will be set depending on what module is loaded at module load time, which could be interesting if you want to swap out java versions, but maybe not what you want.

if this is not what you want I believe you'll have to resort to create an EasyBlock which can get the java environment at module creation time.

File: easybuild/easyconfigs/i/IMOD/IMOD-4.7.15_RHEL6-64_CUDA6.0.eb [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+]
+
+sanity_check_paths = {
+    'files': ['IMOD-linux.sh', 'IMOD-linux.csh', 'installIMOD'],
+    'dirs': ['bin', 'lib']
+}
+
+modextravars = {
+    'IMOD_DIR'                : "$root",
+    'IMOD_JAVADIR'            : os.getenv('EBROOTJAVA'),
```
Comment (boegel) [inline easybuild/easyconfigs/i/IMOD/IMOD-4.7.15_RHEL6-64_CUDA6.0.eb]: hmm, use of `os.getenv` is frowned upon in easyconfigs...

File: easybuild/easyconfigs/i/IMOD/IMOD-4.7.15_RHEL6-64_CUDA6.0.eb (hunk 5 above)
Comment (boegel) [inline easybuild/easyconfigs/i/IMOD/IMOD-4.7.15_RHEL6-64_CUDA6.0.eb]: also, it doesn't work without an `import os` (which makes it worse):

```
NameError: name 'os' is not defined
```

Comment (valtandor) [inline easybuild/easyconfigs/i/IMOD/IMOD-4.7.15_RHEL6-64_CUDA6.0.eb]: I know. I fear an EasyBlock may be required anyway, with a Java dependency being included for good measure. Sigh.

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by valtandor** [comment]
This would seem to be waiting on its EasyBlock.

**Comment by valtandor** [comment]
@boegel There seems to have been a competing EasyBlock uploaded into develop for this version of IMOD. Are you able to comment on that?

**Comment by boegel** [comment]
@valtandor indeed, see #3347; I didn't take this PR into account when merging #3347, my apologies...

It seems that your easyblock @ https://github.com/hpcugent/easybuild-easyblocks/pull/847/files does more than the easyconfig-only approach that got merged does.

So, I suggest we rework this PR to fuse your approach with the one that got merged in with #3347?

I'll send a PR your way to handle that...

**Comment by boegel** [comment]
fusion of your IMOD easyconfig with the one that got merged into `develop` done in https://github.com/valtandor/easybuild-easyconfigs/pull/13

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 2 (1 easyconfigs in this PR)
node2647.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/3d22604007f77131ddf90f6e0901f0aa for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2112.delcatty.os - Linux SL 6.8, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/9b75e330a12d17df6fa7ff3520e5a073 for a full test report.

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Going in, thanks @valtandor!

**Pull request merged**